shell/browser/electron_browser_context.h

```diff
@@ -100,7 +105,14 @@ class ElectronBrowserContext : public content::BrowserContext {
                                       bool in_memory,
                                       base::Value::Dict options = {});
 
+  // Get or create the BrowserContext using the |path|.
+  // The |options| will be passed to constructor when there is no
+  // existing BrowserContext.
+  static ElectronBrowserContext* FromPath(const base::FilePath& path,
+                                          base::Value::Dict options = {});
+
   static BrowserContextMap& browser_context_map();
+  static BrowserContextPathMap& browser_context_path_map();
```

Comment:
Adding another map would require modifying code in some other places, which do things like exit cleanup and context validation, you can search `ElectronBrowserContext::browser_context_map` in the code base to find out them. This is also why CI is failing.

It is possible to duplicate those code to also handle `browser_context_path_map`, but that would make code harder to maintain. One solution is to add `base::FilePath` as a property to `PartitionKey` so we can just put the contexts in one map.

Reply:
> Adding another map would require modifying code in some other places, which do things like exit cleanup and context validation, you can search `ElectronBrowserContext::browser_context_map` in the code base to find out them. This is also why CI is failing.
> 
> It is possible to duplicate those code to also handle `browser_context_path_map`, but that would make code harder to maintain. One solution is to add `base::FilePath` as a property to `PartitionKey` so we can just put the contexts in one map.

Hello @zcbenz 

Thanks for getting back to me. 

Shall I refactor PartitionKey to accept a PartitionOrPath type I added. With the change done, I can ensure that the == and <  comparison operations are done only between matching component types

```
 bool operator==(const PartitionOrPath & other) const {
         if (auto* path_value = std::get_if<std::reference_wrapper<const std::string>>(
          &other)) {
                   const std::string& partition_loc = path_value->get();
                   return (partition == partition_loc ) && (in_memory == other.in_memory);
          } else if (auto* filepath_partition =
                        std::get_if<std::reference_wrapper<const base::FilePath>>( &other)) {
                   const base::FilePath& partition_path = filepath_partition->get();
                   return (partition == partition_path ) ;
         }

 }
```

Do let me know your thoughts.

Thanks,
George.

Reply:
I'm good reusing `PartitionOrPath`, but if we end up using one map to store both kinds of browser contexts, we have to add a new property to `PartitionKey` to distinguish the kinds though.

Reply:
Thansk @zcbenz for getting back to me. I have updated the PR as per your comments and rebased the commits as well. When the final squash is done, the change  should land as a single commit, though I guess the 100s of commits are being shown in the commit section, are only because the branch I was using for this feature went out of date.

Thanks.
George.